### PR TITLE
refactor: remove function out_dm that output the Gamma DM

### LINKED ISSUE #4368: Feature `out_dm` should be deleted, and rename `out_dm1` to `out_dm`. Also feature of `init_dm` is needed.
### Background

In ABACUS, `out_dm` feature only support gamma-only calculation, and only density matrix of gamma_point will be outputted as dense matrix, which is useless now.

I suggest to modify this feature to output real space density matrix (DMR) with non-zero matrix elements as CSR format (the feature of `out_dm1`). 

Feature of `init_dm` for reading DMR file to restart SCF calculation is also needed. 

### Describe the solution you'd like

1. delete code of `out_dm`.
2. refactor code of `out_dm1`, use DensityMatrix::DMR to print CSR format file but not current code.
3. add feature of `init_dm`.

### Task list only for developers

- [ ] Notice possible changes of behavior
- [ ] Explain the changes of codes in core modules of ESolver, HSolver, ElecState, Hamilt, Operator or Psi

### Notice Possible Changes of Behavior (Reminder only for developers)

_No response_

### Notice any changes of core modules (Reminder only for developers)

_No response_

### Notice Possible Changes of Core Modules (Reminder only for developers)

_No response_

### Additional Context

_No response_

### Task list for Issue attackers (only for developers)

- [ ] Review and understand the proposed feature and its importance.
- [ ] Research on the existing solutions and relevant research articles/resources.
- [ ] Discuss with the team to evaluate the feasibility of implementing the feature.
- [ ] Create a design document outlining the proposed solution and implementation details.
- [ ] Get feedback from the team on the design document.
- [ ] Develop the feature following the agreed design.
- [ ] Write unit tests and integration tests for the feature.
- [ ] Update the documentation to include the new feature.
- [ ] Perform code review and address any issues.
- [ ] Merge the feature into the main branch.
- [ ] Monitor for any issues or bugs reported by users after the feature is released.
- [ ] Address any issues or bugs reported by users and continuously improve the feature.

### PR DESCRIPTION
**Pull request by pxlxingliang**
### Reminder
- [ ] Have you linked an issue with this pull request?
- [ ] Have you added adequate unit tests and/or case tests for your pull request?
- [ ] Have you noticed possible changes of behavior below or in the linked issue?
- [ ] Have you explained the changes of codes in core modules of ESolver, HSolver, ElecState, Hamilt, Operator or Psi? (ignore if not applicable)

### Linked Issue
Fix #4368 

### Unit Tests and/or Case Tests for my changes
- A unit test is added for each new feature or bug fix.

### What's changed?
- Example: My changes might affect the performance of the application under certain conditions, and I have tested the impact on various scenarios...

### Any changes of core modules? (ignore if not applicable)
- Example: I have added a new virtual function in the esolver base class in order to ...

### CODE REVIEW

**Comment by pxlxingliang** [comment]
Temporarily remove the function of `out_dm`, and soon I will merge out_dm/out_dm1/out_npz_dm to be one INPUT parameter `out_dm`.

File: source/module_esolver/esolver_ks_lcao.cpp [hunk 1]
```diff
@@ -1108,12 +1107,11 @@ void ESolver_KS_LCAO<TK, TR>::after_scf(const int istep)
...
 
     // 4) write density matrix
-    if (this->LOC.out_dm)
+    if (INPUT.out_dm)
     {
-        for (int is = 0; is < GlobalV::NSPIN; is++)
-        {
-            this->create_Output_DM(is, istep).write();
-        }
+        ModuleBase::WARNING_QUIT("ESolver_KS_LCAO",
```
Comment (mohanchen) [inline source/module_esolver/esolver_ks_lcao.cpp:1112]: It seems that this is a temporary solution, can we use Hcontainer to output density matrix in dense format?

Comment (pxlxingliang) [reply source/module_esolver/esolver_ks_lcao.cpp:1112]: No, we cannot get the dense matrix from HContainer now.

Comment (pxlxingliang) [reply source/module_esolver/esolver_ks_lcao.cpp:1112]: Class DensityMatrix can return the DMK in dense format. But, here is just remove the function of out_dm, we do not need to use HContainer to output the DMK.

**Comment by pxlxingliang** [comment]
close this commit now. Reopen it, after implement the output of DMK based on HContainer.

**Pull request closed without merging**